docs/dev/object-deletion.md

```diff
@@ -0,0 +1,118 @@
+Values other than `true` will result in a CVO failure and should therefore result in CVO CI failure.
+```yaml
+apiVersion: apps/v1
+...
+metadata:
+...
+  annotations:
+    release.openshift.io/delete: "true"
+```
+The existing CVO ordering scheme defined [here](operators.md) is also used for object removal. This provides a simple and familiar method of deleting multiple objects by reversing the order in which they were created. It is the developer's responsibility to ensure proper deletion ordering and to ensure that all items originally created by an object are deleted when that object is deleted. For example, an operator may have to be modified, or a new operator created, to take explicit actions to remove external resources. The modified or new operator would then be removed in a subsequent update.
+
+Similar to how CVO handles create/update requests, deletion requests are implemented in a non-blocking manner whereby CVO issues the initial request to delete an object kicking off resource finalization and after which resource removal. CVO does not wait for actual resource removal but instead continues. CVO logs when a delete is initiated, that the delete is ongoing when a manifest is processed again and found to have a deletion time stamp, and delete completion upon resource finalization.
+
+If an object cannot be successfully removed CVO will set `Upgradeable=False` which in turn blocks cluster update to the next minor release.
+
+## Examples
+
+The following examples provide guidance to OpenShift developers on how resources may be removed but this will vary depending on the component.
+Ultimately it is the developer's responsibility to ensure the removal works by thoroughly testing.
+In all cases, and as general guidance, an operator should never allow itself to be removed if the operator's operand has not been removed.
+
+### The autoscaler operator
+
+Remove the cluster-autoscaler-operator deployment.
+The existing cluster-autoscaler-operator deployment manifest 0000_50_cluster-autoscaler-operator_07_deployment.yaml is modified to contain the delete annotation:
+```yaml
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: cluster-autoscaler-operator
+  namespace: openshift-machine-api
+  annotations:
+    release.openshift.io/delete: "true"
+...
+```
+Additional manifest properties such as `spec` may be set if convenient (e.g. because you are looking to make a minimal change vs. a previous version of the manifest), but those properties have no affect on manifests with the delete annotation.
+
+### The service-catalog operators
+
+In release 4.5 two jobs, openshift-service-catalog-controller-manager-remover and openshift-service-catalog-apiserver-remover, were created to remove the Service Catalog.
+In release 4.6, these jobs and all their supporting cluster objects also needed to be removed.
+This example shows how Service Catalog removal would have been executed had manifest based object deletion been in place.
```

Comment:
Suggestion. Substitute `This example shows how Service Catalog removal would have been executed had manifest based object deletion been in place. ` 
with 
`Here is an example of how service catalog could have been removed using manifest annotation to delete an object `. 

Also I do not this we need to give an example what was the process to remove the object without this feature.

Reply:
reworded